### ISSUE TITLE
Improve Parsing for SDAG Boolean Expressions

### ISSUE THREAD

**Issue by epmikida** [UIUC-PPL/charm#1009]
*Original issue: https://charm.cs.illinois.edu/redmine/issues/1009*

---
See https://lists.cs.illinois.edu/lists/arc/charm/2016-03/msg00003.html for context.

In core meeting, this was discussed and it was stated that in the past Bob's suggested approach was difficult, because even getting the parser to match the parens around an if/while condition was difficult since we can't fully parse C++ but it may be useful to explore this a bit deeper.

**Comment by pplimport** [comment]
*Original author: Robert Steinke*
*Original date: 2016-09-26 16:17:19*

---
In my original request I actually had two suggestions.  The second one may be infeasible if the C++ is hard to parse, but I'd like to draw attention again to my first suggestion.

serial
{
sdagCondition = (some complex boolean expression);
}

if (sdagCondition)
{
...

In order to have the above code in my .ci file I need to have sdagCondition be a member variable of the class.  Could there be some syntactic sugar that even if it doesn't eliminate the need for the serial block, at least eliminates the need for me to declare sdagCondition as a member of my class?  My idea was a hidden member variable declared in the SDAG_CODE macro.
